File: python-sdk/src/foremapi/comments.py
import requests
from typing import Optional
from foremapi.models import shared, operations
from . import utils

class Comments:
    _client: requests.Session
    _security_client: requests.Session
    _server_url: str
    _language: str
    _sdk_version: str
    _gen_version: str

    def __init__(self, client: requests.Session, security_client: requests.Session, server_url: str, language: str, sdk_version: str, gen_version: str) -> None:
        self._client = client
        self._security_client = security_client
        self._server_url = server_url
        self._language = language
        self._sdk_version = sdk_version
        self._gen_version = gen_version

    
    def get_comment_by_id(self, request: operations.GetCommentByIDRequest) -> operations.GetCommentByIDResponse:
        r"""Comment by id
        This endpoint allows the client to retrieve a comment as well as his descendants comments.
        
          It will return the required comment (the root) with its nested descendants as a thread.
        
          See the format specification for further details.
        """
        
        base_url = self._server_url
        
        url = utils.generate_url(base_url, "/api/comments/{id}", request.path_params)
        
        
        client = self._security_client
        
        r = client.request("GET", url)
        content_type = r.headers.get("Content-Type")

        res = operations.GetCommentByIDResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            pass
        elif r.status_code == 404:
            pass

        return res

    
    def get_comments_by_article_id(self, request: operations.GetCommentsByArticleIDRequest) -> operations.GetCommentsByArticleIDResponse:
        r"""Comments
        This endpoint allows the client to retrieve all comments belonging to an article or podcast episode as threaded conversations.
        
        It will return the all top level comments with their nested comments as threads. See the format specification for further details.
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/api/comments"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = self._security_client
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.GetCommentsByArticleIDResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[list[shared.Comment]])
                res.comments = out
        elif r.status_code == 404:
            pass

        return res

    

File: ts-sdk/src/sdk/comments.ts
import { AxiosInstance, AxiosRequestConfig, AxiosResponse, ParamsSerializerOptions } from "axios";
import * as operations from "./models/operations";
import * as utils from "../internal/utils";

export class Comments {
  _defaultClient: AxiosInstance;
  _securityClient: AxiosInstance;
  _serverURL: string;
  _language: string;
  _sdkVersion: string;
  _genVersion: string;

  constructor(defaultClient: AxiosInstance, securityClient: AxiosInstance, serverURL: string, language: string, sdkVersion: string, genVersion: string) {
    this._defaultClient = defaultClient;
    this._securityClient = securityClient;
    this._serverURL = serverURL;
    this._language = language;
    this._sdkVersion = sdkVersion;
    this._genVersion = genVersion;
  }
  
  /**
   * getCommentById - Comment by id
   *
   * This endpoint allows the client to retrieve a comment as well as his descendants comments.
   * 
   *   It will return the required comment (the root) with its nested descendants as a thread.
   * 
   *   See the format specification for further details.
  **/
  getCommentById(
    req: operations.GetCommentByIdRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.GetCommentByIdResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.GetCommentByIdRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = utils.generateURL(baseURL, "/api/comments/{id}", req.pathParams);
    
    const client: AxiosInstance = this._securityClient!;
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...config,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.GetCommentByIdResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            break;
          case httpRes?.status == 404:
            break;
        }

        return res;
      })
  }

  
  /**
   * getCommentsByArticleId - Comments
   *
   * This endpoint allows the client to retrieve all comments belonging to an article or podcast episode as threaded conversations.
   * 
   * It will return the all top level comments with their nested comments as threads. See the format specification for further details.
  **/
  getCommentsByArticleId(
    req: operations.GetCommentsByArticleIdRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.GetCommentsByArticleIdResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.GetCommentsByArticleIdRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/api/comments";
    
    const client: AxiosInstance = this._securityClient!;
    
    const qpSerializer: ParamsSerializerOptions = utils.getQueryParamSerializer(req.queryParams);

    const requestConfig: AxiosRequestConfig = {
      ...config,
      params: req.queryParams,
      paramsSerializer: qpSerializer,
    };
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...requestConfig,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.GetCommentsByArticleIdResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `application/json`)) {
                res.comments = httpRes?.data;
            }
            break;
          case httpRes?.status == 404:
            break;
        }

        return res;
      })
  }

}
